src/stt_data_with_llm/audio_parser.py

```diff
@@ -1,6 +1,270 @@
+import io
+import logging
+import os
+
+import librosa
+import requests
+import torchaudio
+from pyannote.audio import Pipeline
+from pydub import AudioSegment
+
+from stt_data_with_llm.config import (
+    AUDIO_SEG_LOWER_LIMIT,
+    AUDIO_SEG_UPPER_LIMIT,
+    HEADERS,
+    HYPER_PARAMETERS,
+)
+from stt_data_with_llm.util import setup_logging
+
+# Call the setup_logging function at the beginning of your script
+setup_logging("audio_parser.log")
+
+
+def sec_to_millis(seconds):
+    """Converts seconds to milliseconds.
+
+    Args:
+        seconds (float): Time in seconds
+
+    Returns:
+        float: Time in milliseconds
+    """
+    return seconds * 1000
+
+
+def frame_to_sec(frame, sampling_rate):
+    """Converts audio frames to seconds based on sampling rate.
+
+    Args:
+        frame (int): audio frame number
+        sampling_rate (int): Audio sampling rate in Hz
+
+    Returns:
+        int: Time in seconds
+    """
+    return frame / sampling_rate
+
+
+def sec_to_frame(sec, sr):
+    """Converts seconds to frame number based on sampling rate.
+
+    Args:
+        sec (float): Time in seconds
+        sr (int): Audio sampling rate in Hz
+
+    Returns:
+        float: Frame number
+    """
+    return sec * sr
+
+
+pipeline = Pipeline.from_pretrained(
+    "pyannote/voice-activity-detection",
+    use_auth_token="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",
```

Comment:
if the key is something sensitive u need to import it using environment variable. else add the key in config and import